src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToInterpolatedString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal partial class ConvertToInterpolatedStringRefactoringProvider :
+        AbstractConvertToInterpolatedStringRefactoringProvider
+            <InterpolatedStringExpressionSyntax,
+             InterpolatedStringContentSyntax,
+             InterpolationSyntax,
+             InvocationExpressionSyntax,
+             ExpressionSyntax,
+             ArgumentSyntax,
+             LiteralExpressionSyntax>
+    {
+        protected override SeparatedSyntaxList<ArgumentSyntax>? GetArguments(InvocationExpressionSyntax invocation) =>
+            invocation?.ArgumentList.Arguments;
+
+        protected override InterpolatedStringExpressionSyntax GetInterpolatedString(string text) =>
+            ParseExpression("$" + text) as InterpolatedStringExpressionSyntax;
+
+        protected override string GetText(LiteralExpressionSyntax literalExpression)
+        {
+            var stringToken = literalExpression.Token;
+            var text = stringToken.ToString();
+
+            if (stringToken.IsVerbatimStringLiteral())
+            {
+                // We need to escape braces as this is an ambiguous case in interpolated strings that is 
+                // not ambiguous in verbatim strings
+                text = text.Replace(@"'{'", @"'{{'").Replace(@"'}'", @"'}}'");
+            }
+
+            return text;
+        }
+
+        protected override bool IsStringLiteral(LiteralExpressionSyntax firstArgument) =>
+            firstArgument?.IsKind(SyntaxKind.StringLiteralExpression) == true;
+
+        protected override SyntaxList<InterpolatedStringContentSyntax> GetInterpolatedStringContents(InterpolatedStringExpressionSyntax interpolatedString) =>
+            interpolatedString.Contents;
+
+        protected override SyntaxNode ReplaceExpressionInInterpolation(InterpolationSyntax node, ExpressionSyntax expressionSyntax) => 
```

Comment:
Could be lifted up to the base type.  Just call.  "node.ReplaceNode(GetExpressionOfInterpolation(), expressionSyntax)"